src/deserialize_uri.rs

```diff
@@ -11,6 +11,9 @@ impl<'de> de::Visitor<'de> for UriVisitor {
     }
 
     fn visit_str<E: de::Error>(self, val: &str) -> Result<Self::Value, E> {
+        #[allow(clippy::map_err_ignore)]
+        // Allowing because the intention is to ignore the previous error, and it would
+        // be impossible to implement one foreign trait on another
         val.parse()
             .map_err(|_| de::Error::invalid_value(de::Unexpected::Str(val), &self))
```

Comment:
Does clippy pass if we do this?

```suggestion
        val.parse()
            .map_err(|_err| de::Error::invalid_value(de::Unexpected::Str(val), &self))
```

Reply:
Yes it does, changing to this